articles/exception-handling-in-javascript/index.md

```diff
@@ -0,0 +1,159 @@
+
+### JavaScript Error Types
+Different errors may occur while executing a JavaScript code. There are three types of errors.
+
+1. [**Syntax Errors:**](https://developer.mozilla.org/en-US/docs/Glossary/Syntax_error#:~:text=An%20exception%20caused%20by%20the,you%20trigger%20a%20syntax%20error.) errors that cannot be interpreted by the computer. These errors stop the program from working. In JavaScript, these errors are:
+
+- Spelling errors(wrong spelling such as fiction instead of function)
+- The omission of important characters such as not using a semicolon to end a statement
+- Use of the wrong indentation
+
+2. [**Runtime Errors:**](https://www.geeksforgeeks.org/javascript-error-and-exceptional-handling-with-examples/#:~:text=Runtime%20Error%3A%20A%20runtime%20error,also%20known%20as%20the%20exceptions.&amp;text=try%20___%20catch%20method%3A%20JavaScript,operator%20to%20handle%20the%20exception.)these errors take place during execution. The errors are detected when your program runs. It crashes or raises an exception. Thus, exception handlers handle exception errors. These errors are often caused by:
+
+- The program cannot find data because it does not exist.
+- It cannot act on the data because it is an invalid type of data.
+
+3. [**Logical Errors:**](https://study.com/academy/lesson/errors-in-javascript-types-methods-prevention.html) these are errors that take place due to logical mistakes. These types of errors do not throw an error or an exception at all. But do not do what the developer intends the code to do. It is challenging to find logical errors. They can only be found through extensive testing.
+
+### Error Objects
+When a runtime error occurs, it stops the code and raises an error object. The error object has two properties:
+
+- Name: It gives the error name.
+- Message: It sets or returns the error message in the form of a string.
+
+JavaScript uses six types of error objects. These error objects are the foundation of exception handling.
+
+- **EvalError**: The `EvalError` function indicates the error that occurred in the [`eval()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval) function. It is a global function that evaluates the JavaScript string. This exception is not thrown anymore by JavaScript. It is used for compatibility.
+- **RangeError**: A `RangeError` is raised when a numeric value is outside the specified range.
+- **ReferenceError**: A `ReferenceError` exception is raised if a variable that is not declared is used. These exceptions commonly occur due to spelling errors on variables.
+- **Syntax Error**: A Syntax Error exception is raised while compiling a code with syntax errors.
+- **TypeError**: A `TypeError` exception is raised when a value is different from the one expected.
+- **URIError**: A `URIError` exception is raised by `encodeURI()` and `decodeURI()` methods.
```

Comment:
It would be useful to the reader if you could provide the MDN links of these two methods...